cohortextractor/tpp_backend.py

```diff
@@ -1019,6 +1019,103 @@ def _patients_with_events(
 
         return codelist_queries + extra_queries + [sql]
 
+    def patients_with_these_decision_support_values(self, **kwargs):
+        # First, we pop the kwargs.
+        algorithm = kwargs.pop("algorithm")
+        # TODO: What is replacing e.g. `on_or_before` with `between`?
```

Comment:
Useful, thank you. For consistency, should I move the following to `process_arguments`? That seems to be the best place to raise a `ValueError`.

https://github.com/opensafely-core/cohort-extractor/blob/4cfe606b28d94d849eb8ce99de86c9699408cd2e/cohortextractor/tpp_backend.py#L1033-L1038

I would make `algorithm_to_pk` a constant called `DECISION_SUPPORT_ALGORITHM_TO_PK` because we would need the mapping in both `process_arguments` and `patients_with_these_decision_support_values`.

Reply:
No, I'd keep the mapping here in the method definition as it's specific to this method. `process_arguments` is for generic things which apply across all methods.

Reply:
Sounds good. Removed comment from source.